app/components/agenda/agendaitem/agendaitem-decision.js

```diff
@@ -73,31 +74,34 @@ export default class AgendaitemDecisionComponent extends Component {
     });
     await piece.save();
     this.args.treatment.set('report', piece);
-    this.args.treatment.save();
+    await this.args.treatment.save();
     this.isAddingReport = false;
     await this.loadReport.perform();
   }
 
-  @task
-  *attachNewReportVersion(piece) {
-    yield piece.save();
+  @action
+  async attachNewReportVersion(piece) {
+    await piece.previousPiece;
```

Comment:
I don't understand why we have to fetch this relation here.

Reply:
I added this somewhere during testing, the `loadReport` task was not loading the `previousPiece` when it should have.
Changed some more thing later that probably fixed it. This await is no longer needed, just tested it again without.